gui/wxpython/core/settings.py

```diff
@@ -25,18 +25,51 @@
 import sys
 import copy
 import wx
+import json
 
 from core import globalvar
 from core.gcmd import GException, GError
 from core.utils import GetSettingsPath, PathJoin, rgb2str
 
 
+class SettingsJSONEncoder(json.JSONEncoder):
+    """Custom JSON encoder"""
+    def default(self, obj):
+        """Encode not automatically serializable objects"""
+        if isinstance(obj, (wx.FontFamily, wx.FontStyle, wx.FontWeight)):
+            return int(obj)
+        return json.JSONEncoder.default(self, obj)
+
+    def iterencode(self, obj):
+        """Encode color tuple"""
```

Comment:
Please, add info on why this is needed, i.e., why the internal representation cannot match plain JSON either expressed as a list of three or a hex code.